src/nwm_routing/src/nwm_routing/output.py

```diff
@@ -213,6 +214,24 @@ def nwm_output_generator(
         LOG.debug("writing CSV file took %s seconds." % (time.time() - start))
         # usgs_df_filtered = usgs_df[usgs_df.index.isin(csv_output_segments)]
         # usgs_df_filtered.to_csv(output_path.joinpath("usgs_df.csv"))
+        
+    if chano:
+        
+        LOG.info("- writing t-route flow results at gage locations to CHANOBS file")
+        start = time.time()
+                
+        nhd_io.write_chanobs(
```

Comment:
One issue I saw with this method is that if the file already exists it will just be appended after the first time running. If I continue to rerun a yaml it will continuously grow the output file. Maybe it would be better to delete the file if it exists before writing to it once again?

Reply:
I spoke with our NCAR colleagues and they confirmed that appending to existing files is the best way to go. Say a simulation composed of 10 loops fails half way through the fifth. We wouldn't want to throw out the results of the first 4 loops and start from the beginning. Instead, we can use restart files to launch a follow up simulation from the top of the fifth loop and continue appending to the same netCDF output file. Yes, this will duplicate some information in the netCDF, but the user can filter-our non unique dates on when they are analyzing information in the netCDF.